app/services/hyrax/edit_permissions_service.rb

```diff
@@ -0,0 +1,178 @@
+    # * yields a PermissionPresenter to provide additional logic and text for rendering
+    #
+    # @param permission_hash [Hash<:name, :access>]
+    # @return false if the given permission_hash is a fixed exclusion
+    # @yield PermissionPresenter
+    #
+    # @see #excluded_permission?
+    def with_applicable_permission(permission_hash:)
+      return false if excluded_permission?(permission_hash)
+      yield(PermissionPresenter.new(service: self, permission_hash: permission_hash))
+    end
+
+    # @api private
+    #
+    # A helper class to contain specific presentation logic related to
+    # the EditPermissionsService
+    class PermissionPresenter
+      # @param service [Hyrax::EditPermissionsService]
+      # @param permission_hash [Hash]
+      def initialize(service:, permission_hash:)
+        @service = service
+        @permission_hash = permission_hash
+      end
+
+      # A hint at how permissions are granted.
+      #
+      # @return String
+      # rubocop:disable Rails/OutputSafety
+      def granted_by_html_hint
+        html = ""
+        @service.unauthorized_collection_managers.each do |managers|
+          next unless name == managers.fetch(:name)
+          html += "<br />Access granted via collection #{managers.fetch(:id)}"
+        end
+        html.html_safe
+      end
+      # rubocop:enable Rails/OutputSafety
+
+      # @return String
+      def name
+        @permission_hash.fetch(:name)
+      end
+
+      # @return String
+      def access
+        @permission_hash.fetch(:access)
+      end
+
+      # @return Boolean
+      # @see EditPermissionsService#cannot_edit_permissions?
+      def can_edit?
+        !@service.cannot_edit_permissions?(@permission_hash)
+      end
+    end
+
+    private
+
+    # Fixed set of users & groups to exclude from "editable" section of display
+    def exclude_from_display
+      [::Ability.public_group_name, ::Ability.registered_group_name, ::Ability.admin_group_name, @depositor]
+    end
+
+    BlockedPermissions = Struct.new(:unauthorized_managers, :unauthorized_collection_managers)
+
+    # find all of the other managers of collections which a user cannot manage
+    #
+    #   Process used:
+    #   - find all of the work's collections which a user can manage
+    #   - find all of the work's collections (of a type which shares permissions) that a user cannot manage
+    #   - find all of the managers of these collections the user cannot manage
+    #   This gives us the manager permissions the user is not authorized to update.
+    #
+    # @return [Struct] BlockedPermissions
+    #   - unauthorized_managers [Array] ids of managers of all collections
+    #   - unauthorized_collection_managers [Array hashes] manager ids & collection_ids [{:name, :id}]
+    def manager_permissions_to_block
+      unauthorized_managers = []
+      unauthorized_collection_managers = []
+      if object_unauthorized_collection_ids.any?
```

Comment:
can this check be dropped? `object_unauthorized_collection_ids.each` will be a no-op if it's empty, yes?